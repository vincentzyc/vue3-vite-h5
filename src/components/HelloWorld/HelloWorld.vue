<template>
  <div>{{ count }} x {{ times }} = {{ result }}</div>
  <van-button type="primary" @click="times += 1">x1</van-button>
</template>

<script setup lang="ts">
import { computed, ref } from "vue";

const props = defineProps<{ count: number }>();

const times = ref(2);
const result = computed(() => props.count * times.value);

defineExpose(props);
</script>
