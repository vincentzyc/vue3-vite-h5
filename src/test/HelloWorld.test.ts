import { mount } from '@vue/test-utils'
import { Lazyload } from "vant"
import HelloWorld from '@/components/HelloWorld.vue'

test('hello world mount', async () => {
  expect(HelloWorld).toBeTruthy()

  const wrapper = mount(HelloWorld, {
    props: {
      count: 4,
    },
    global: {
      directives: {
        lazy: Lazyload as any
      }
    },
  })

  expect(wrapper.text()).toContain('4 x 2 = 8')
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.get('button').trigger('click')

  expect(wrapper.text()).toContain('4 x 3 = 12')

  await wrapper.get('button').trigger('click')

  expect(wrapper.text()).toContain('4 x 4 = 16')
})