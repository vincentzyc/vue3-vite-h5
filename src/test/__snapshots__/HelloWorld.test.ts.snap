// Vitest Snapshot v1

exports[`hello world mount 1`] = `
"<div>4 x 2 = 8</div>
<button type=\\"button\\" class=\\"van-button van-button--primary van-button--normal\\">
  <div class=\\"van-button__content\\">
    <!----><span class=\\"van-button__text\\">x1</span>
    <!---->
  </div>
</button>"
`;
